src/webviews/extension-side/dataviewer/dataViewerDependencyService.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { SemVer } from 'semver';
+import { ProductNames } from '../../../kernels/installer/productNames';
+import { Product } from '../../../kernels/installer/types';
+import { IApplicationShell } from '../../../platform/common/application/types';
+import { traceWarning } from '../../../platform/logging';
+import { IsCodeSpace } from '../../../platform/common/types';
+import { parseSemVer } from '../../../platform/common/utils';
+import { DataScience, Common } from '../../../platform/common/utils/localize';
+import { EnvironmentType, PythonEnvironment } from '../../../platform/pythonEnvironments/info';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { IDataViewerDependencyService } from './types';
+import { executeSilently } from '../../../kernels/helpers';
+import { IJupyterSession, IKernel, IKernelProvider } from '../../../kernels/types';
+
+const minimumSupportedPandaVersion = '0.20.0';
+
+function isVersionOfPandasupported(version: SemVer) {
+    return version.compare(minimumSupportedPandaVersion) > 0;
+}
+
+/**
+ * Responsible for managing dependencies of a Data Viewer.
+ */
+@injectable()
+export class DataViewerDependencyService implements IDataViewerDependencyService {
+    private get kernel(): IKernel {
+        const kernel = this.kernelProvider.kernels.find((kernel) => kernel.status === 'idle');
+        if (!kernel) {
+            sendTelemetryEvent(Telemetry.NoIdleKernel);
+            throw new Error(DataScience.noIdleKernel());
+        }
+        return kernel;
+    }
+    private get kernelSession(): IJupyterSession {
+        if (!this.kernel.session) {
+            sendTelemetryEvent(Telemetry.NoActiveKernelSession);
+            throw new Error(DataScience.noActiveKernelSession());
+        }
+        return this.kernel.session;
+    }
+    private get packaging(): string {
+        const envType = this.kernel.kernelConnectionMetadata.interpreter?.envType;
+        const isConda = envType === EnvironmentType.Conda;
+        return isConda ? 'conda' : 'pip';
+    }
+
+    constructor(
+        @inject(IApplicationShell) private readonly applicationShell: IApplicationShell,
+        @inject(IKernelProvider) private readonly kernelProvider: IKernelProvider,
+        @inject(IsCodeSpace) private isCodeSpace: boolean
+    ) {}
+
+    public async checkAndInstallMissingDependencies(interpreter: PythonEnvironment): Promise<void> {
```

Comment:
You'd pass the IKernel here instead of the environment.

Reply:
Or maybe even a function that can execute code and return the results. Then it wouldn't need to know about debuggers or kernels.

Reply:
Yes, this is what I suggested to @sadasant when I spoke to him about this, I think I didn't explain it well enough.

Reply:
Now I'm passing IKernel! Thank you for your guidance 🙏